api/src/main/java/org/apache/iceberg/encryption/KeyManagementClient.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.encryption;
+
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+/** A minimum client interface to connect to a key management service (KMS). */
+public interface KeyManagementClient extends Serializable {
```

Comment:
I'm not yet convinced that this should be different than the `EncryptionManager`. We should discuss in more detail, but why can't we add key encryption/decryption to the existing interface?

We already have a way to inject that through the Table API, which would allow us to keep reflection code in the catalogs and also easily pass this functionality around.